Made Legion Dependency on Gasnet Tarball Explicit

rspavel opened the pull request:
Currently the legion build system fetches a known compatible version of gasnet, builds it itself, and uses it as a dependency. To do this, it fetches the file as part of the cmake command

This PR makes that dependency explicit in the form of a `resource()` so that it will be captured in spack mirrors for the purposes of building and deploying on machines with restrictive firewalls that may block direct access to github

rspavel committed:
Made Legion Dependency on Gasnet Tarball Explicit

Made legion dependency on a gasnet tarball explicit so as to take
advantage of spack mirrors for the purpose of deploying on machines with
firewalls

streichler reviewed (commented):
The changes look reasonable to me.  I'd like @pmccormick to have a chance to look at
it as well though.

rspavel committed:
Re-Ran `spack style`

Thought I caught all of the style errors but it wouldn't be a PR without
an extra commit to pass style checks.

pmccormick commented:
The changes LGMT.   Green light for the upstreaming from me once the checks complete.
